storage/store.go

```diff
@@ -2506,6 +2511,8 @@ func (s *Store) processRaft() {
 			s.pendingRaftGroups.Unlock()
 			s.mu.Unlock()
 
+			s.processRaftMu.Lock()
```

Comment:
Yeah, though that's probably too much (and complicated too) - but we to somehow make sure a replica isn't removed from the Store before we go and let it handle Raft.

It should be enough to ensure that a Replica is put in a deleted state before removing it from the store though, as long as handleRaftReady can deal with that. We don't currently have that though, hence my anxiety about this gap.